### PR TITLE
Implement metrics chart component to overview and monitoring page

### PR DESCRIPTION
**Pull request by YiniXu9506**
**What this PR does?**

- apply metrics chart component.
- resolve y-axis tick format when unit is `short` and tick value is less than 0.1.

### CODE REVIEW

File: ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/Monitoring/context.ts [hunk 1]
```diff
@@ -9,22 +9,15 @@ import client from '~/client'
...
-      },
-      options
-    )
+  metricsQueryGet(params: {
+    endTimeSec?: number
+    query?: string
+    startTimeSec?: number
+    stepSec?: number
+  }) {
+    return client.getInstance().metricsQueryGet({ ...params }, {
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/Monitoring/context.ts:18]: `metricsQueryGet(params)` should be ok.

File: ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/Overview/context.ts [hunk 2]
```diff
@@ -20,22 +20,15 @@ class DataSource implements IOverviewDataSource {
...
-      },
-      options
-    )
+  metricsQueryGet(params: {
+    endTimeSec?: number
+    query?: string
+    startTimeSec?: number
+    stepSec?: number
+  }) {
+    return client.getInstance().metricsQueryGet({ ...params }, {
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/Overview/context.ts:29]: ditto

File: ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/context.ts [hunk 3]
```diff
@@ -9,22 +9,15 @@ import client from '~/client'
...
-      },
-      options
-    )
+  metricsQueryGet(params: {
+    endTimeSec?: number
+    query?: string
+    startTimeSec?: number
+    stepSec?: number
+  }) {
+    return client.getInstance().metricsQueryGet({ ...params }, {
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/context.ts:18]: ditto

**Review by shhdgit — APPROVED** [review]

File: ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx [hunk 4]
```diff
@@ -52,6 +51,19 @@ export default function Monitoring() {
...
+  const handleOnBrush = (range: TimeRangeValue) => {
+    setChartRange(range)
+  }
+
+  const ErrorComponent = (error: Error) => (
+    <Space direction="vertical">
+      <ErrorBar errors={[error]} />
+      <Link to="/user_profile?blink=profile.prometheus">
+        {t('components.metricChart.changePromButton')}
+      </Link>
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx:63]: in `tidb-dashboard-for-dbaas` and `tidb-dashboard-for-clinic-op` variant, we don't have `/user_profile` route.

File: ui/packages/tidb-dashboard-lib/src/apps/Overview/components/Metrics.tsx [hunk 5]
```diff
@@ -50,6 +48,19 @@ export default function Metrics() {
...
+  const handleOnBrush = (range: TimeRangeValue) => {
+    setChartRange(range)
+  }
+
+  const ErrorComponent = (error: Error) => (
+    <Space direction="vertical">
+      <ErrorBar errors={[error]} />
+      <Link to="/user_profile?blink=profile.prometheus">
+        {t('components.metricChart.changePromButton')}
+      </Link>
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Overview/components/Metrics.tsx:60]: ditto, not all variant have `/user_profile` route.

**Review by shhdgit — APPROVED** [review]

**Review by baurine — APPROVED** [review]

**Pull request merged**